pkg/controller/node/ipam/cloud_cidr_allocator.go

```diff
@@ -42,84 +45,149 @@ import (
 // deallocation is delegated to the external provider, and the controller
 // merely takes the assignment and updates the node spec.
 type cloudCIDRAllocator struct {
-	lock sync.Mutex
-
 	client clientset.Interface
 	cloud  *gce.GCECloud
-
-	recorder record.EventRecorder
+	// Channel that is used to pass updating Nodes with assigned CIDRs to the background
+	// This increases a throughput of CIDR assignment by not blocking on long operations.
+	nodeCIDRUpdateChannel chan nodeAndCIDR
+	recorder              record.EventRecorder
```

Comment:
insert a newline here to make the nodesInProcessing a separate block

Reply:
Done. Addressed this and all further comments in both this file and range_allocator.go.